cmd/whitesourceExecuteScan.go

```diff
@@ -184,65 +314,202 @@ func resolveProjectIdentifiers(config *ScanOptions, utils whitesourceUtils, sys
+	case "mta":
+		// Execute scan for maven and all npm modules
+		if err := executeMTAScan(config, scan, utils); err != nil {
 			return err
 		}
-	default:
-		// Download the unified agent jar file if one does not exist
-		if err := downloadAgent(config, utils); err != nil {
+	case "maven":
+		// Execute scan with maven plugin goal
+		if err := executeMavenScan(config, scan, utils); err != nil {
 			return err
 		}
-
-		// Auto generate a config file based on the working directory's contents.
-		// TODO/NOTE: Currently this scans the UA jar file as a dependency since it is downloaded beforehand
-		if err := autoGenerateWhitesourceConfig(config, utils); err != nil {
+	case "npm":
+		// Execute scan with in each npm module using npm.Executor
+		if err := executeNpmScan(config, scan, utils); err != nil {
 			return err
 		}
-
-		// Execute whitesource scan with unified agent jar file
-		if err := executeUAScan(config, utils); err != nil {
+	case "yarn":
+		// Execute scan with whitesource yarn plugin
+		if err := executeYarnScan(config, scan, utils); err != nil {
+			return err
+		}
+	default:
+		// Execute scan with Unified Agent jar file
+		if err := executeUAScan(config, scan, utils); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
 // executeUAScan executes a scan with the Whitesource Unified Agent.
-func executeUAScan(config *ScanOptions, utils whitesourceUtils) error {
+func executeUAScan(config *ScanOptions, scan *whitesourceScan, utils whitesourceUtils) error {
+	// Download the unified agent jar file if one does not exist
+	if err := downloadAgent(config, utils); err != nil {
+		return err
+	}
+
+	// Auto generate a config file based on the working directory's contents.
+	// TODO/NOTE: Currently this scans the UA jar file as a dependency since it is downloaded beforehand
+	if err := autoGenerateWhitesourceConfig(config, utils); err != nil {
+		return err
+	}
+
 	return utils.RunExecutable("java", "-jar", config.AgentFileName, "-d", ".", "-c", config.ConfigFilePath,
-		"-apiKey", config.OrgToken, "-userKey", config.UserToken, "-project", config.ProjectName,
+		"-apiKey", config.OrgToken, "-userKey", config.UserToken, "-project", scan.aggregateProjectName,
 		"-product", config.ProductName, "-productVersion", config.ProductVersion)
 }
 
+// executeMTAScan executes a scan for the Java part with maven, and performs a scan for each NPM module.
+func executeMTAScan(config *ScanOptions, scan *whitesourceScan, utils whitesourceUtils) error {
+	log.Entry().Infof("Executing Whitesource scan for MTA project")
+	err := executeMavenScanForPomFile(config, scan, utils, "pom.xml")
+	if err != nil {
+		return err
+	}
+
+	return executeNpmScan(config, scan, utils)
+}
+
+// executeMavenScan constructs maven parameters from the given configuration, and executes the maven goal
+// "org.whitesource:whitesource-maven-plugin:19.5.1:update".
+func executeMavenScan(config *ScanOptions, scan *whitesourceScan, utils whitesourceUtils) error {
```

Comment:
What about extracting this and related functions to an own package `pkg/whitesource/maven`?